torch/csrc/inductor/aoti_runtime/sycl_runtime_wrappers.h

```diff
@@ -0,0 +1,173 @@
+  const char* build_flags = "";
+  const ze_module_format_t format = ZE_MODULE_FORMAT_IL_SPIRV;
+  ze_module_desc_t module_description = {};
+  module_description.stype = ZE_STRUCTURE_TYPE_MODULE_DESC;
+  module_description.format = format;
+  module_description.inputSize = binary_size;
+  module_description.pInputModule = (uint8_t*)binary_ptr;
+  module_description.pBuildFlags = build_flags;
+  ze_module_build_log_handle_t buildlog;
+  ze_module_handle_t module;
+  auto context_initial = context;
+  auto device_initial = device;
+  auto error_no = ZE_RESULT_SUCCESS;
+  error_no =
+      zeModuleCreate(context, device, &module_description, &module, &buildlog);
+
+  if (error_no != ZE_RESULT_SUCCESS) {
+    size_t szLog = 0;
+    ZE_CHECK(zeModuleBuildLogGetString(buildlog, &szLog, nullptr));
+    char* strLog = (char*)malloc(szLog);
+    ZE_CHECK(zeModuleBuildLogGetString(buildlog, &szLog, strLog));
+    std::cerr << "L0 build module failed. Log: " << strLog << std::endl;
+    free(strLog);
+    ZE_CHECK(zeModuleBuildLogDestroy(buildlog));
+  }
+  ZE_CHECK(error_no);
+  return module;
+}
+
+ze_kernel_handle_t create_function(
+    ze_module_handle_t module,
+    ze_kernel_flags_t flag,
+    std::string func_name) {
+  ze_kernel_handle_t kernel;
+  ze_kernel_desc_t kernel_description = {};
+  kernel_description.stype = ZE_STRUCTURE_TYPE_KERNEL_DESC;
+  kernel_description.pNext = nullptr;
+  kernel_description.flags = flag;
+  kernel_description.pKernelName = func_name.c_str();
+  assert(module);
+  ZE_CHECK(zeKernelCreate(module, &kernel_description, &kernel));
+  return kernel;
+}
+
+static ze_module_handle_t loadModule(
+    sycl::queue& queue,
+    std::string& spv_path) {
+  auto l0_device = sycl::get_native<sycl::backend::ext_oneapi_level_zero>(
+      queue.get_device());
+  auto l0_context = sycl::get_native<sycl::backend::ext_oneapi_level_zero>(
+      queue.get_context());
+  std::ifstream IFS(spv_path.c_str(), std::ios::binary);
+  std::ostringstream OSS;
+  OSS << IFS.rdbuf();
+  std::string data(OSS.str());
+
+  return create_module(
+      l0_context,
+      l0_device,
+      reinterpret_cast<const uint8_t*>(data.c_str()),
+      data.size());
+}
+
+static sycl::kernel* getKernel(
+    sycl::queue& queue,
+    ze_module_handle_t l0_module,
+    const char* kernel_name) {
+  assert(l0_module);
+  assert(kernel_name);
+  auto l0_kernel =
+      create_function(l0_module, ZE_KERNEL_FLAG_FORCE_RESIDENCY, kernel_name);
+  auto sycl_device = queue.get_device();
+  auto ctx = sycl_device.get_platform().ext_oneapi_get_default_context();
+
+  auto mod = sycl::make_kernel_bundle<
+      sycl::backend::ext_oneapi_level_zero,
+      sycl::bundle_state::executable>(
+      {l0_module, sycl::ext::oneapi::level_zero::ownership::transfer}, ctx);
+
+  auto fun = sycl::make_kernel<sycl::backend::ext_oneapi_level_zero>(
+      {mod, l0_kernel, sycl::ext::oneapi::level_zero::ownership::transfer},
+      ctx);
+  auto kernel_ptr = new sycl::kernel(fun);
```

Comment:
May I know when and where `kernel_ptr` will be deallocated?

Reply:
resolved, thanks for your reminder.